records_mover/records/pandas/prep_for_csv.py

```diff
@@ -56,7 +57,10 @@ def _convert_series_or_index(series_or_index: T,
                                  'timeonlyformat',
                                  records_format.hints)
                 pandas_time_format = '%H:%M:%S'
-            if isinstance(series_or_index, pd.Series):
+
+            if isinstance(series_or_index[0], datetime.time):
+                return series_or_index.apply(lambda d: d.strftime(pandas_time_format))
+            elif isinstance(series_or_index, pd.Series):
```

Comment:
After cleaning up the tests of a bunch of pandas-related workarounds, it appears that time columns don't always get formatted correctly in Pandas.  This was discovered here in the integration tests:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1854/workflows/344dba04-cf95-4c46-b1d6-bdbc53ff3ad9/jobs/15884

I wrote a component test further down the test pyramid, which failed this way, which the above code fixed:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1855/workflows/dc6a566c-413f-43a0-ae1c-6b2b114f0b40/jobs/15894